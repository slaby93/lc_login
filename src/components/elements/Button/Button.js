import styled, { css } from 'styled-components';
import { prop, ifProp } from 'styled-tools';

const StyledComponent = styled.button`
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px solid ${prop('theme.color.black')};
  font-size: 16px;
  background-color: ${prop('theme.color.white')};
  border-radius: 2px;
  border-color: ${prop('theme.color.dirtyWhite')};
  padding: 10px;
  color:  ${prop('color', prop('theme.color.black'))};
  background-color:  ${prop('bgColor', prop('theme.color.white'))};
  min-height: 50px;
  box-shadow: 0px 1px 8px 1px ${prop('theme.color.whiteShadow')};
  transition: all .5s ease-out, opacity .5s ease-out;
  outline: none;
  cursor: pointer;
  &:active {
    box-shadow: 2px 2px 7px 0px ${prop('theme.color.dirtyShadow')};
  }
  &:hover:not(:active) {
    transform: scale(1.03);
  }
  ${ifProp('grow', css`
      width: 100%;
  `)}
  ${ifProp('disabled', css`
     cursor: not-allowed;
     opacity: 0.4;
  `)}
  ${ifProp('round', css`
     border-radius: 50%;
  `)}
  ${ifProp('text', css`
      display: inline;
      border: none;
      box-shadow: none;
      margin-bottom: 10px;
      height: initial;
      margin: 0;
      width: fit-content;
      &:active {
        box-shadow:none;
      }
      &:hover {
        transform: scale(1.05);
      }
  `)}

  ${ifProp('noBorder', css`
    border: none;
  `)}
`;

export default StyledComponent;
