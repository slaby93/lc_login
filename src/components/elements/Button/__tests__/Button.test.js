import React from 'react';
import { shallow } from 'enzyme';
import Button from './../Button';

describe('<Button />', () => {
  test('CSS', () => {
    const mountedComponent = shallow(<Button
      group="1"
      value="a"
      field="test"
      className="test"
    >
      TEST
    </Button>);
    expect(mountedComponent).toMatchSnapshot();
  });

  describe('props', () => {
    test('grow', () => {
      const mountedComponent = shallow(<Button
        group="1"
        value="a"
        field="test"
        className="test"
        grow
      >
        TEST
      </Button>);
      expect(mountedComponent).toMatchSnapshot();
    });

    test('disabled', () => {
      const mountedComponent = shallow(<Button
        group="1"
        value="a"
        field="test"
        className="test"
        disabled
      >
        TEST
      </Button>);
      expect(mountedComponent).toMatchSnapshot();
    });

    test('round', () => {
      const mountedComponent = shallow(<Button
        group="1"
        value="a"
        field="test"
        className="test"
        round
      >
        TEST
      </Button>);
      expect(mountedComponent).toMatchSnapshot();
    });

    test('text', () => {
      const mountedComponent = shallow(<Button
        group="1"
        value="a"
        field="test"
        className="test"
        text
      >
        TEST
      </Button>);
      expect(mountedComponent).toMatchSnapshot();
    });

    test('noBorder', () => {
      const mountedComponent = shallow(<Button
        group="1"
        value="a"
        field="test"
        className="test"
        noBorder
      >
        TEST
      </Button>);
      expect(mountedComponent).toMatchSnapshot();
    });
  });
});
