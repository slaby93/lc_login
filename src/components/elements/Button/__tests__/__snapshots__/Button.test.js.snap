// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> CSS 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid;
  font-size: 16px;
  border-radius: 2px;
  padding: 10px;
  min-height: 50px;
  box-shadow: 0px 1px 8px 1px;
  -webkit-transition: all .5s ease-out,opacity .5s ease-out;
  transition: all .5s ease-out,opacity .5s ease-out;
  outline: none;
  cursor: pointer;
}

.c0:active {
  box-shadow: 2px 2px 7px 0px;
}

.c0:hover:not(:active) {
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

<button
  className="test c0"
  value="a"
>
  TEST
</button>
`;

exports[`<Button /> props disabled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid;
  font-size: 16px;
  border-radius: 2px;
  padding: 10px;
  min-height: 50px;
  box-shadow: 0px 1px 8px 1px;
  -webkit-transition: all .5s ease-out,opacity .5s ease-out;
  transition: all .5s ease-out,opacity .5s ease-out;
  outline: none;
  cursor: pointer;
  cursor: not-allowed;
  opacity: 0.4;
}

.c0:active {
  box-shadow: 2px 2px 7px 0px;
}

.c0:hover:not(:active) {
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

<button
  className="test c0"
  disabled={true}
  value="a"
>
  TEST
</button>
`;

exports[`<Button /> props grow 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid;
  font-size: 16px;
  border-radius: 2px;
  padding: 10px;
  min-height: 50px;
  box-shadow: 0px 1px 8px 1px;
  -webkit-transition: all .5s ease-out,opacity .5s ease-out;
  transition: all .5s ease-out,opacity .5s ease-out;
  outline: none;
  cursor: pointer;
  width: 100%;
}

.c0:active {
  box-shadow: 2px 2px 7px 0px;
}

.c0:hover:not(:active) {
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

<button
  className="test c0"
  value="a"
>
  TEST
</button>
`;

exports[`<Button /> props noBorder 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid;
  font-size: 16px;
  border-radius: 2px;
  padding: 10px;
  min-height: 50px;
  box-shadow: 0px 1px 8px 1px;
  -webkit-transition: all .5s ease-out,opacity .5s ease-out;
  transition: all .5s ease-out,opacity .5s ease-out;
  outline: none;
  cursor: pointer;
  border: none;
}

.c0:active {
  box-shadow: 2px 2px 7px 0px;
}

.c0:hover:not(:active) {
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

<button
  className="test c0"
  value="a"
>
  TEST
</button>
`;

exports[`<Button /> props round 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid;
  font-size: 16px;
  border-radius: 2px;
  padding: 10px;
  min-height: 50px;
  box-shadow: 0px 1px 8px 1px;
  -webkit-transition: all .5s ease-out,opacity .5s ease-out;
  transition: all .5s ease-out,opacity .5s ease-out;
  outline: none;
  cursor: pointer;
  border-radius: 50%;
}

.c0:active {
  box-shadow: 2px 2px 7px 0px;
}

.c0:hover:not(:active) {
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

<button
  className="test c0"
  value="a"
>
  TEST
</button>
`;

exports[`<Button /> props text 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 1px solid;
  font-size: 16px;
  border-radius: 2px;
  padding: 10px;
  min-height: 50px;
  box-shadow: 0px 1px 8px 1px;
  -webkit-transition: all .5s ease-out,opacity .5s ease-out;
  transition: all .5s ease-out,opacity .5s ease-out;
  outline: none;
  cursor: pointer;
  display: inline;
  border: none;
  box-shadow: none;
  margin-bottom: 10px;
  height: initial;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0:active {
  box-shadow: 2px 2px 7px 0px;
}

.c0:hover:not(:active) {
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

.c0:active {
  box-shadow: none;
}

.c0:hover {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

<button
  className="test c0"
  value="a"
>
  TEST
</button>
`;
